public/app/core/components/NavBar/NavBarDropdown.tsx

```diff
@@ -47,72 +48,80 @@ const NavBarDropdown = ({
   return (
     <ul className={`${styles.menu} dropdown-menu dropdown-menu--sidemenu`} role="menu">
       <li>{header}</li>
-      {items
-        .filter((item) => !item.hideFromMenu)
-        .map((child, index) => (
-          <DropdownChild
-            key={`${child.url}-${index}`}
-            isDivider={child.divider}
-            icon={child.icon as IconName}
-            onClick={child.onClick}
-            target={child.target}
-            text={child.text}
-            url={child.url}
-          />
-        ))}
+      {filteredItems.map((child, index) => (
+        <DropdownChild
+          key={`${child.url}-${index}`}
+          isDivider={child.divider}
+          icon={child.icon as IconName}
+          onClick={child.onClick}
+          target={child.target}
+          text={child.text}
+          url={child.url}
+        />
+      ))}
       {subtitleText && <li className={styles.subtitle}>{subtitleText}</li>}
     </ul>
   );
 };
 
 export default NavBarDropdown;
 
-const getStyles = (theme: GrafanaTheme2, reverseDirection: Props['reverseDirection']) => ({
-  header: css`
-    background-color: ${theme.colors.background.secondary};
-    border: none;
-    color: ${theme.colors.text.primary};
-    font-size: ${theme.typography.h4.fontSize};
-    font-weight: ${theme.typography.h4.fontWeight};
-    padding: ${theme.spacing(1)} ${theme.spacing(1)} ${theme.spacing(1)} ${theme.spacing(2)} !important;
-    white-space: nowrap;
-    width: 100%;
+const getStyles = (
+  theme: GrafanaTheme2,
+  reverseDirection: Props['reverseDirection'],
+  filteredItems: Props['items']
+) => {
+  const hasNoChildren = filteredItems!.length === 0;
```

Comment:
nit: I can see this variable adjusts the height of the header but I'm not sure why that is from the var name. Is it compensating for a border somewhere? Don't know if there's a better name for it's purpose. 🤔

Reply:
yep exactly, it's compensating for the border on the whole menu which i don't really like tbh, but i can't think of a better way to do it 😕  i can definitely adjust the name though 👍